<launch>
    <node name="dynamixel_manager" pkg="dynamixel_controllers" type="controller_manager.py" required="true" output="screen">
        <rosparam>
            namespace: dxl_manager
            serial_ports:
                actuator_port:
                    port_name: "/dev/ttyUSB0"
                    baud_rate: 57600
                    min_motor_id: 1
                    max_motor_id: 50
                    update_rate: 15
        </rosparam>
    </node>
    
	    <!-- Load controller configuration to parameter server -->
    <rosparam file="$(find robot_connect)/launch/config/dynamixel_joint_controllers.yaml" command="load"/>
    
    <!-- start specified joint controllers -->
    <node name="dynamixel_controller_spawner" pkg="dynamixel_controllers" type="controller_spawner.py"
          args="--manager=dxl_manager
                --port=actuator_port
                --type=simple
                pan_kinect_2
                tilt_kinect_2"
          output="screen"/>
  
    <!--node name="tilt_laser_control" pkg="robot_connect" type="tilt_laser_control" output="screen"/-->
    <node name="pan_tilt_height_control" pkg="robot_connect" type="pan_tilt_height_control.py">
        <remap from="/pan_kinect/state" to="/pan_kinect_2/state"/>
        <remap from="/tilt_kinect/state" to="/tilt_kinect_2/state"/>
    </node>
    <!--node pkg="robot_connect" type="kinect_preprocess" name="kinect_preprocess" output="screen"/-->
    <include file="$(find robot_connect)/launch/kinect_launch.launch"/>
</launch>
